classes/task/ProcessQueueJobs.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @file classes/task/ProcessQueueJobs.inc.php
+ *
+ * Copyright (c) 2022 Simon Fraser University
+ * Copyright (c) 2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ProcessQueueJobs
+ * @ingroup tasks
+ *
+ * @brief Class to process queue jobs via the schedular task
+ */
+
+namespace PKP\task;
+
+use PKP\scheduledTask\ScheduledTask;
+
+class ProcessQueueJobs extends ScheduledTask
+{
+    /**
+     * @copydoc ScheduledTask::getName()
+     */
+    public function getName()
+    {
+        return __('admin.scheduledTask.processQueueJobs');
+    }
+
+
+    /**
+     * @copydoc ScheduledTask::executeActions()
+     */
+    public function executeActions()
+    {
+        $jobQueue = app('pkpJobQueue');
+
+        $jobBuilder = $jobQueue->getJobModelBuilder();
+
+        if ($jobBuilder->count() <= 0) {
+            return true;
+        }
+
+        while ($jobBuilder->count()) {
+            $jobQueue->runJobsInQueue();
+        }
+
+        return true;
+    }
+}
+
+if (!PKP_STRICT_MODE) {
+    class_alias('\PKP\task\ProcessQueueJobs', '\ProcessQueueJobs');
```

Comment:
New classes don't need aliases; this is just for backwards compatibility.

Reply:
But without this I am getting errors as 

```
[08-Aug-2022 17:39:15 Asia/Dhaka] PHP Fatal error:  Uncaught Exception: Cannot instantiate class. Class "ReviewReminder" is not declared in "lib.pkp.classes.task.ReviewReminder". in /Users/abir/Sites/code/ojs-main/lib/pkp/includes/functions.php:223
Stack trace:
#0 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/cliTool/ScheduledTaskTool.php(117): instantiate('lib.pkp.classes...', NULL, NULL, 'execute', Array)
#1 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/cliTool/ScheduledTaskTool.php(103): PKP\cliTool\ScheduledTaskTool->executeTask('lib.pkp.classes...', Array)
#2 /Users/abir/Sites/code/ojs-main/lib/pkp/classes/cliTool/ScheduledTaskTool.php(73): PKP\cliTool\ScheduledTaskTool->parseTasks('registry/schedu...')
#3 /Users/abir/Sites/code/ojs-main/tools/runScheduledTasks.php(25): PKP\cliTool\ScheduledTaskTool->execute()
#4 {main}
  thrown in /Users/abir/Sites/code/ojs-main/lib/pkp/includes/functions.php on line 223
```
Similar errors for any schedule tasks . May be I am missing something .

Reply:
I think it should be possible to specify a fully qualified class name in registry/scheduledTasks.xml, in place of the `dot.separated.class.names` we have relied upon prior to adding classloading support.

Reply:
providing the fully qualified namespace also does (for example : `\PKP\task\ProcessQueueJobs` or `PKP\task\ProcessQueueJobs`) not resolve this as throwing exception as such 

```
Cannot instantiate task class.
```

originated from https://github.com/pkp/pkp-lib/blob/b376a074f127e39b737c07787e1ccbf9cfc053ca/classes/cliTool/ScheduledTaskTool.php#L118 . The core of that lies in the defined regex here at https://github.com/pkp/pkp-lib/blob/b376a074f127e39b737c07787e1ccbf9cfc053ca/includes/functions.php#L178 which returning a `false` . 

I think it will be better to set up a new issue and update the code with backward compatibility to handle `.` and `/` class paths and update the schedule tasks as a PR of that issue . That way we can also do bit clean up for the schedule tasks also .

Reply:
Can you try this again without using the `class_alias`? It should be possible to just use the FQCN (per my comment on the OJS repo: https://github.com/pkp/ojs/pull/3476/files#r951951685) and if it's not working then it means I have to fix something over on pkp/pkp-lib#6091. This has probably been fixed since your last comment a week ago -- my apologies!

Reply:
It's working fine now . Fixed .